certbot/certbot/_internal/main.py

```diff
@@ -1359,8 +1360,12 @@ def set_displayer(config):
     """
     if config.quiet:
         config.noninteractive_mode = True
+
+        devnull = open(os.devnull, "w")
+        atexit.register(devnull.close)
```

Comment:
What do you think about transforming `set_displayer` into a context manager, instead of using atexit ?

Reply:
Sure, I gave it a shot.

I don't mind one way or another, because it's process-scoped in the end. One upside is that once we delete `IReporter`, we'll have zero `atexit` usages.